cmake_ext.py

```diff
@@ -128,22 +129,27 @@ def build_extension(self, ext):
         if cmake_cache.exists():
             with cmake_cache.open() as f:
                 s = f.read()
-                cached_generator = cache_value("CMAKE_GENERATOR", s)
-                cached_cfg = cache_value("CMAKE_BUILD_TYPE", s)
-                disagreement = []
-                for arg in cmake_args:
-                    if arg.startswith("-D"):
-                        key, value = arg[2:].split("=")
-                        cached_value = cache_value(key, s)
-                        disagreement.append(value != cached_value)
-                if any(
-                    disagreement
-                    + [
-                        (cmake_generator and cached_generator != cmake_generator),
-                        cached_cfg != cfg,
-                    ]
-                ):
-                    cmake_cache.unlink()
+            cached_generator = cache_value("CMAKE_GENERATOR", s)
+            cached_cfg = cache_value("CMAKE_BUILD_TYPE", s)
+            disagreement = []
+            for arg in cmake_args:
+                if arg.startswith("-D"):
+                    key, value = arg[2:].split("=")
+                    cached_value = cache_value(key, s)
+                    # Change \ to / in case of Windows
+                    disagreement.append(
+                        PureWindowsPath(value).as_posix()
+                        != PureWindowsPath(cached_value).as_posix()
+                    )
```

Comment:
That does not look right. There should be no need to import and use PureWindowsPath, PurePath on Windows uses PureWindowsPath automatically. It should be sufficient PurePath(value).as_posix() != PurePath(cached_value).as_posix()

Reply:
I applied this fix, let's see whether that works as intended.

Reply:
If I remember correctly, the problem was that CMake converts paths so that next they are not recognized. I saw paths like `...\site-packages\numpy\f2py/src` with mixed slashes. The idea was to bring all paths to the form where only forward slashes present. I checked `Path(value).absolute()` doesn't do that, so `Path("\\numpy\\f2py/src").absolute()` is not equal to `Path("/numpy/f2py/src").absolute()`. I don't know whether it is critical, as I cannot reproduce the problem that was intended to fix this way.

Reply:
FYI, `.absolute()` is undocumented. `.resolve()` is the correct function to use, but be warned, there's a bug (fixed in Python 3.10, but present in all older versions) where it will not resolve to an absolute path if the path doesn't exist on Windows. It's fine if the path does exist, and you can manually use `Path.cwd() / <path>` before resolving to enforce the absoluteness of the resolved path.